ui/fancy/src/js/autorepeat.js

```diff
@@ -0,0 +1,53 @@
+class KeyAutorepeater {
+    constructor(repeatDelay, repeatRate) {
+      this.repeatDelay = repeatDelay;
+      this.repeatRate = repeatRate;
+      this.listeners = new Map();
+      this.repeaters = new Set();
+      this.keyDownData = new Map();
+      window.addEventListener('keydown', (ev) => {
+        if (!this.listeners.has(ev.key)) {
+          return;
+        }
+        ev.preventDefault();
+        if (this.keyDownData.has(ev.key)) {
+          return;
+        }
+        let id = null;
```

Comment:
This should be on first line of the function. And there is no need to set it `null`.